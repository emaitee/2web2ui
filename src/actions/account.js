import sparkpostApiRequest from 'src/actions/helpers/sparkpostApiRequest';

export function fetch(params = {}) {
  return sparkpostApiRequest({
    type: 'FETCH_ACCOUNT',
    meta: {
      method: 'GET',
      url: '/account',
      params
    }
  });
}

export function getPlans() {
  return sparkpostApiRequest({
    type: 'GET_PLANS',
    meta: {
      method: 'GET',
      url: '/account/plans'
    }
  });
}

export function update(data) {
  return sparkpostApiRequest({
    type: 'UPDATE_ACCOUNT',
    meta: {
      method: 'PUT',
      url: '/account',
      data
    }
  });
}


export function register(data) {

  return sparkpostApiRequest({
    type: 'CREATE_ACCOUNT',
    meta: {
      method: 'POST',
      url: '/account',
      data
    }
  });
}
