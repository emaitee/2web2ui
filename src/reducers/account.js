const initialState = {
  updateError: null,
  loading: false,
  subscription: {},
  updateLoading: false,
  createLoading: false
};

export default (state = initialState, { type, meta, payload }) => {
  switch (type) {
    case 'FETCH_ACCOUNT_PENDING': {
      return { ...state, loading: true };
    }

    case 'FETCH_ACCOUNT_SUCCESS':
      return { ...state, loading: false, ...payload };

    case 'FETCH_ACCOUNT_FAIL':
      return { ...state, loading: false };

    case 'UPDATE_ACCOUNT_PENDING':
      return { ...state, updateLoading: true, updateError: null };

    case 'UPDATE_ACCOUNT_SUCCESS':
      return { ...state, updateLoading: false, ...meta.data };

    case 'UPDATE_ACCOUNT_FAIL':
      return { ...state, updateError: payload, updateLoading: false };

    case 'CREATE_ACCOUNT_PENDING':
      return { ...state, createLoading: true, createError: null };

    case 'CREATE_ACCOUNT_SUCCESS':
      return { ...state, createLoading: false, payload };

    case 'CREATE_ACCOUNT_FAIL':
      return { ...state, createError: payload, updateLoading: false };

    default:
      return state;
  }
};
