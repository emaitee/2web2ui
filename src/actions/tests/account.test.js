import { fetch, getPlans, register } from '../account';
jest.mock('../helpers/sparkpostApiRequest', () => jest.fn((a) => a));

test('fetch - no params', () => {
  const fetchAction = fetch();
  expect(fetchAction).toMatchSnapshot();
});

test('fetch with params', () => {
  const params = {
    this: 'one',
    also: 'that one'
  };
  const fetchAction = fetch(params);
  expect(fetchAction).toMatchSnapshot();
});

test('getPlans', () => {
  const getPlansAction = getPlans();
  expect(getPlansAction).toMatchSnapshot();
});

describe('Account action creators', () => {
  describe('register', () => {
    let data;
    beforeEach(() => {
      data = {
        first_name: 'foo',
        last_name: 'bar',
        email: 'foo@bar.com',
        passowrd: 'foobar',
        tou_accepted: true,
        recaptcha_response: 'foofoo',
        recaptcha_type: 'invisible'
      };
    });
    it('makes api call with passed data', () => {
      expect(register(data)).toMatchSnapshot();
    });
  });

});
