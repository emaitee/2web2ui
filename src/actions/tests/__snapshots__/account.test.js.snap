// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account action creators register makes api call with passed data 1`] = `
Object {
  "meta": Object {
    "data": Object {
      "email": "foo@bar.com",
      "first_name": "foo",
      "last_name": "bar",
      "passowrd": "foobar",
      "recaptcha_response": "foofoo",
      "recaptcha_type": "invisible",
      "tou_accepted": true,
    },
    "method": "POST",
    "url": "/account",
  },
  "type": "CREATE_ACCOUNT",
}
`;

exports[`fetch - no params 1`] = `
Object {
  "meta": Object {
    "method": "GET",
    "params": Object {},
    "url": "/account",
  },
  "type": "FETCH_ACCOUNT",
}
`;

exports[`fetch with params 1`] = `
Object {
  "meta": Object {
    "method": "GET",
    "params": Object {
      "also": "that one",
      "this": "one",
    },
    "url": "/account",
  },
  "type": "FETCH_ACCOUNT",
}
`;

exports[`getPlans 1`] = `
Object {
  "meta": Object {
    "method": "GET",
    "url": "/account/plans",
  },
  "type": "GET_PLANS",
}
`;
